docs/virtual-disk-troubleshoot.md

```diff
@@ -0,0 +1,138 @@
+    Warning  Update  7s (x2 over 13s)  spc-controller  reconcile is disabled via "reconcile.openebs.io/disable" annotation
+  
+  ```
+
+- Update the CSP with correct blockdevice details. Replace the old blockdevices with new blockdevices.
+  ```sh
+  $ kubectl edit csp cstor-pool-yfn3 
+  apiVersion: openebs.io/v1alpha1
+  kind: CStorPool
+  metadata:
+    annotations:
+      openebs.io/csp-lease: '{"holder":"openebs/cstor-pool-yfn3-6bbc79f5fd-9dlmb","leaderTransition":1}'
+    creationTimestamp: "2020-08-03T10:45:22Z"
+    finalizers:
+    - openebs.io/pool-protection
+    .....
+    .....
+  spec:
+    group:
+    - blockDevice:
+      - deviceID: /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol06591ee2acf1e6419
+        inUseByPool: true
+        name: blockdevice-8e782e6b47ed896a325870fc436e65f9
+    - blockDevice:
+      - deviceID: /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol09dfbbc322ed0647d
+        inUseByPool: true
+        name: blockdevice-99596b3a4eb2396b8e45b334daeefccc
+  
+  cstorpool.openebs.io/cstor-pool-yfn3 edited
+  ```
+  After editing the CSP successfully cross verify whether changes have been saved using following command and it show output as updated blockdevice name
+  ```sh
+  $ kubectl get csp cstor-pool-yfn3 -o yaml | grep blockdevice
+        name: blockdevice-8e782e6b47ed896a325870fc436e65f9
+        name: blockdevice-99596b3a4eb2396b8e45b334daeefccc
+  ```
+
+- Update the blockdevice entry on SPC. This step is required to make it intact with CSP since we manually updated the blockdevice name in CSP.
+  ```sh
+  $ kubectl edit spc cstor-pool
+  apiVersion: openebs.io/v1alpha1
+  kind: StoragePoolClaim
+  metadata:
+    annotations:
+      reconcile.openebs.io/disable: "true"
+    creationTimestamp: "2020-08-03T10:45:22Z"
+    finalizers:
+    - storagepoolclaim.openebs.io/finalizer
+    .....
+    .....
+  spec:
+    blockDevices:
+      blockDeviceList:
+      - blockdevice-8e782e6b47ed896a325870fc436e65f9
+      - blockdevice-99596b3a4eb2396b8e45b334daeefccc
+  ```
+  After editing the SPC we are making sure that CSP and SPC are pointing to the same blockdevices and cStor pool is also running on top of mentioned devices. Verify whether it saved successfully or not using following command
+  ```sh
+  $ kubectl get spc cstor-pool -o yaml | grep blockdevice-
+      - blockdevice-8e782e6b47ed896a325870fc436e65f9
+      - blockdevice-99596b3a4eb2396b8e45b334daeefccc
+  ```
+
+- Enable the reconciliation on SPC. This step is required to enable back the reconciliation on SPC which was stopped by
+adding `reconcile.openebs.io/disable annotation` on SPC. Enabling can be achieved by removing annotation on SPC using `kubectl edit` command. Then remove the annotation `reconcile.openebs.io/disable` which was added in above steps. This will help to reconcile further changes in SPC if made. After removing the annotation there shouldn’t be any more events generated for SPC saying disabled reconciliation.
```

Comment:
After modifying will it be good to rerun the script before upgrading/migrating?

Reply:
IMO it is not required as after upgrade this situation will not occur.